r/vignettes/developing.Rmd

```diff
@@ -38,567 +38,31 @@ set -e
-* Using {withr}'s `with_makevars(list(LDFLAGS = ""), ...)`
-* adding `LDFLAGS=` to your `~/.R/Makevars` file (the least recommended way, though it is a common debugging approach suggested online)
-
-```{bash, save=run & !sys_install & macos, hide=TRUE}
-# Setup troubleshooting section
-# install a system-level arrow on macOS
-brew install apache-arrow
-```
-
-
-```{bash, save=run & !sys_install & ubuntu, hide=TRUE}
-# Setup troubleshooting section
-# install a system-level arrow on Ubuntu
-sudo apt update
-sudo apt install -y -V ca-certificates lsb-release wget
-wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
-sudo apt install -y -V ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
-sudo apt update
-sudo apt install -y -V libarrow-dev
-```
-
-```{bash, save=run & !sys_install & macos}
-MAKEFLAGS="LDFLAGS=" R CMD INSTALL .
-```
-
-
-## `rpath` issues
-
-If the package fails to install/load with an error like this:
-
-```
-  ** testing if installed package can be loaded from temporary location
-  Error: package or namespace load failed for 'arrow' in dyn.load(file, DLLpath = DLLpath, ...):
-  unable to load shared object '/Users/you/R/00LOCK-r/00new/arrow/libs/arrow.so':
-  dlopen(/Users/you/R/00LOCK-r/00new/arrow/libs/arrow.so, 6): Library not loaded: @rpath/libarrow.14.dylib
-```
-
-ensure that `-DARROW_INSTALL_NAME_RPATH=OFF` was passed (this is important on
-macOS to prevent problems at link time and is a no-op on other platforms).
-Alternatively, try setting the environment variable `R_LD_LIBRARY_PATH` to
-wherever Arrow C++ was put in `make install`, e.g. `export
-R_LD_LIBRARY_PATH=/usr/local/lib`, and retry installing the R package.
-
-When installing from source, if the R and C++ library versions do not
-match, installation may fail. If you've previously installed the
-libraries and want to upgrade the R package, you'll need to update the
-Arrow C++ library first.
-
-For any other build/configuration challenges, see the [C++ developer
-guide](https://arrow.apache.org/docs/developers/cpp/building.html).
-
-## Other installation issues
-
-There are a number of scripts that are triggered when the arrow R package is installed. For package users who are not interacting with the underlying code, these should all just work without configuration and pull in the most complete pieces (e.g. official binaries that we host). However, knowing about these scripts can help package developers troubleshoot if things go wrong in them or things go wrong in an install.  See [the installation vignette](./install.html#how-dependencies-are-resolved) for more information.
+* where necessary add extra arguments to the function signature for features 
+that don't exist in R but do in Arrow (e.g. passing in a schema when reading a
+CSV dataset)
```

Comment:
I think this vignette needs more after here but I don't know exactly what.  Maybe something on writing bindings between compute kernels and R functions? Or is that a bit too specific?

Reply:
I think an example of a compute kernel binding would be good. We could use that as an example to further explain the above points.

I'd also suggest linking to recent PRs that may serve an examples. I find a reference PR particularly helpful in reminding me which files I might need to modify.

Reply:
Great point, agreed on the reference PR.  I'm also gonna tag some of the wider R dev team to pitch in on this, as I can't help but feeling there's a bit more to discuss on what docs we distribute with the package vs. what docs we just have on the pkgdown site.

Reply:
@jonkeane What are your thoughts on this?

Reply:
Personally, I think of the pkgdown site as *the* canonical documentation, especially for vignettes. So I don't have a strong opinion / I'm not worried about not including it inside of / distributed along side the package.

As for examples: I think that would be great. We can link to PRs (though if the link goes to CRAN it's susceptible to a redirect/rot that will anger CRAN — we should and do check for that, but just a reminder). Though sometimes when writing examples I find it a little bit easier to make a dedicated example that has lots of extra comments/commentary/possibly even glosses over some of the reality involved with implementing them.